library/sensu_check_dict.py

```diff
@@ -44,7 +44,11 @@ def main():
                     module.params['name']: module.params['check']
                 }
             })
-
+            only_on_ip = check['checks'][module.params['name']].pop('only_on_ip', '')
+            if only_on_ip !='':
+                command = 'if ip a | grep ' +  only_on_ip + ' > /dev/null 2>&1; then ' 
+                check['checks'][module.params['name']]['command']=             \
+                   command + check['checks'][module.params['name']]['command'] + '; fi'
```

Comment:
check dependency is to reduce alert notification noise only,  it cannot reduce the metrics.
https://sensuapp.org/docs/0.23/enterprise/filters/check-dependencies.html

and using shell script is what it is in current ursula code, like this:
https://github.com/blueboxgroup/ursula/blob/master/roles/neutron-data/tasks/monitoring.yml#L14-L22
https://github.com/blueboxgroup/ursula/blob/master/roles/neutron-data/templates/check-neutron-l3-routers.sh
Let me think about if there are a better way to realize it, and if you have better options, let me know.